pkg/nix/cmd/build.go

```diff
@@ -13,10 +15,49 @@ func Build(dir string, attribute string) error {
 	}
 	cmd := exec.Command("nix", "build", attribute, "-o", dir)
 
-	cmd.Stdout = os.Stdout
-	// TODO: in future- we can pipe to stderr pipe and modify error messages to be understandable by the user
-	cmd.Stderr = os.Stderr
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		return err
+	}
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return err
+	}
+	stdErrChan:= make(chan string)
 
+	go func(){
+		scanner:= bufio.NewScanner(stderr)
```

Comment:
Can we please make a dedicated function for this instead of inlining? And add a unit test for it?

Reply:
Similar with below go routine